src/components/Booking.jsx

```diff
@@ -0,0 +1,82 @@
+import React, { useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import DatePicker from 'react-datepicker';
+import 'react-datepicker/dist/react-datepicker.css';
+import { format } from 'date-fns';
+import { Select, Option } from '@material-tailwind/react';
+import { bookCar } from '../redux/Reservations/reservationsSlice';
+import api from '../api/api';
+import { useAuthUser } from '../redux/Auth/useAuthUser';
+import { allCars } from '../redux/Home/home';
+
+const Booking = () => {
+  const [pickupDate, setPickupDate] = useState(null);
+  const [returnDate, setReturnDate] = useState(null);
+  const user = {
+    email: 'abc@gmail.com',
+    password: '123456',
+  };
+  api.login(user);
+  const cars = useSelector(allCars);
+  const [carId, setCarId] = useState('');
+  const currentUser = useAuthUser();
+  const dispatch = useDispatch();
+
+  const handleCarId = (carId) => {
+    setCarId(parseInt(carId, 10));
+  };
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    const reservation = {
+      pickup_date: format(pickupDate, 'yyyy-MM-dd'),
+      return_date: format(returnDate, 'yyyy-MM-dd'),
+      car_id: carId,
+    };
+    const reservationObject = {
+      reservation,
+      user_id: currentUser.id,
+    };
+    dispatch(bookCar(reservationObject));
+  };
```

Comment:
Please check it again this isn't working when I submit this form it gives me a 500 status error so kindly fix this also. :man_technologist: 

![Screenshot from 2023-01-11 12-38-52](https://user-images.githubusercontent.com/45226848/211745972-2b8f0e31-d925-4ea4-b3a7-4385e600da75.png)

- This error I see in console:
![Screenshot from 2023-01-11 12-38-32](https://user-images.githubusercontent.com/45226848/211746019-da7e58f0-e650-4539-9472-4ed00fc6a96d.png)

- This error I see in the network.
![Screenshot from 2023-01-11 12-42-15](https://user-images.githubusercontent.com/45226848/211746951-091d02c2-7fcf-40b8-9753-b97c9d6ccea7.png)

### Also check your design when I select any date it's not come in the date picker field.

Reply:
This has been resolved the problem is from @aamirkhan2478's localhost